src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs

```diff
@@ -423,5 +406,17 @@ protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e
 		}
 
 		protected virtual void UpdateBackgroundColor() => BackgroundColor = Element.BackgroundColor.ToUIColor();
+
+		protected void DisposeObservers(ref IDisposable disposable)
+		{
+			disposable?.Dispose();
+			disposable = null;
+		}
+
+		protected void DisposeObservers(ref NSObject disposable)
+		{
+			disposable?.Dispose();
+			disposable = null;
+		}
```

Comment:
can we leave only the first method? (IDisposable) ?
**DisposeObserver** sounds better, do you agree?

Reply:
Some observers are NSObject and do the same code, said that I think that's good to have the `NSObject` overload

Reply:
Yes) But NSObject implements IDisposable, otherwise you wouldn't have the `Dispose` method)

Reply:
Yes, but if you try to use the `ref IDisposable` method the compiler will complain.
![image](https://user-images.githubusercontent.com/20712372/101260404-62b48680-370e-11eb-8c9d-550014557192.png)

Reply:
Ah, okay. Interesting)
Then maybe let's store all observers as NSObjects instead IDisposable?

Reply:
It's better to have it as `IDisposable` when I need to use a `key` to assign the observer, because the `AddObserver` method returns `IDisposable` as you can [see here](https://docs.microsoft.com/en-us/dotnet/api/Foundation.NSObject.AddObserver?view=xamarin-ios-sdk-12#Foundation_NSObject_AddObserver_System_String_Foundation_NSKeyValueObservingOptions_System_Action_Foundation_NSObservedChange__) on the docs.

Reply:
But we cast it to NSObject in code now

Why cannot we preserve them as NSObject? It worked before :)
```csharp
rateObserver = (NSObject)avPlayerViewController.Player.AddObserver("rate", NSKeyValueObservingOptions.New, ObserveRate);
```

Reply:
I missed that cast, we don't need it. I don't about the history of this code, so I guess that the original author didn't know that it's ok to have this as `IDisposable`.